fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt

```diff
@@ -96,13 +98,15 @@ internal class HttpClient(
 
             setDoOutput(connection, request.method)
             setBodyIfDoOutput(connection, request)
+            stethoHook?.preConnect(connection, request)
```

Comment:
I prefer `hooks.preConnect`, where `hooks` takes multiple hooks, not just one.

Are you sure THIS is where `HttpURLConnection actually connects? The body here has been sent already, so it seems that connection happens way earlier.